infrastructure/src/main/java/org/corfudb/infrastructure/log/LogMetadata.java

```diff
@@ -28,12 +31,16 @@
     @Getter
     private volatile long globalTail;
 
+    @Getter
+    private final Map<UUID, StreamAddressSpace> streamsAddressSpaceMap;
+
     @Getter
     private final Map<UUID, Long> streamTails;
 
     public LogMetadata() {
         this.globalTail = Address.NON_ADDRESS;
         this.streamTails = new HashMap();
+        this.streamsAddressSpaceMap = new HashMap();
```

Comment:
`this.streamsAddressSpaceMap = new HashMap<>();`

Reply:
done.